api/src/main/java/org/openmrs/api/db/hibernate/HibernateDiagnosisDAO.java

```diff
@@ -100,18 +104,65 @@ public Diagnosis saveDiagnosis(Diagnosis diagnosis) {
 	}
 
 	/**
-	 * Gets primary diagnoses for a given encounter
+	 * Gets diagnoses for an Encounter. When specified, this method only returns
```

Comment:
What does this mean? `When specified, this method only returns primary or confirmed diagnoses`?

Reply:
I was trying to express that optionally this method can only return primary or confirmed diagnoses. It's not very clear though so I removed the phrase. I think the arguments and the rest of the docstring make it clear enough how the method is to be used.